libcontainerd/client_linux.go

```diff
@@ -470,16 +463,33 @@ func (clnt *client) Restore(containerID string, attachStdio StdioCallback, optio
 				logrus.Warnf("libcontainerd: restore was called on a fully synced container (%s)", containerID)
 				return nil
 			}
-			// the container is not running so we need to fix the state within docker
-			ev = &containerd.Event{
-				Type:   StateExit,
-				Status: 1,
+		} else if ev.Pid != InitFriendlyName || ev.Type != StateExit {
```

Comment:
duh! 

Well, that leave only the case where the event logs was rotated then, and the call to `getContainer` becomes useless.